lib/src/client.dart

```diff
@@ -793,12 +793,20 @@ class Client {
   }
 
   /// Closes the websocket connection and resets the client
-  Future<void> disconnect({bool flushOfflineStorage = false}) async {
+  Future<void> disconnect({
+    bool flushOfflineStorage = false,
+    bool clearUser = false,
+  }) async {
     logger.info('Disconnecting');
 
     await _offlineStorage?.disconnect(flush: flushOfflineStorage);
     _offlineStorage = null;
 
+    if (clearUser == true) {
```

Comment:
What about a log here? Might be useful for debugging purposes. At the very least, we can log the options the function was called with.